File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/git_diff_summary.py
# coding: utf-8
# Copyright (c) 2016, 2026, Oracle and/or its affiliates.  All rights reserved.
# This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.


from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class GitDiffSummary(object):
    """
    Summary information about a file that differs in a git folder context.
    """

    #: A constant which can be used with the change_type property of a GitDiffSummary.
    #: This constant has a value of "ADDED"
    CHANGE_TYPE_ADDED = "ADDED"

    #: A constant which can be used with the change_type property of a GitDiffSummary.
    #: This constant has a value of "MODIFIED"
    CHANGE_TYPE_MODIFIED = "MODIFIED"

    #: A constant which can be used with the change_type property of a GitDiffSummary.
    #: This constant has a value of "DELETED"
    CHANGE_TYPE_DELETED = "DELETED"

    #: A constant which can be used with the change_type property of a GitDiffSummary.
    #: This constant has a value of "RENAMED"
    CHANGE_TYPE_RENAMED = "RENAMED"

    #: A constant which can be used with the change_type property of a GitDiffSummary.
    #: This constant has a value of "COPIED"
    CHANGE_TYPE_COPIED = "COPIED"

    #: A constant which can be used with the change_type property of a GitDiffSummary.
    #: This constant has a value of "TYPE_CHANGED"
    CHANGE_TYPE_TYPE_CHANGED = "TYPE_CHANGED"

    #: A constant which can be used with the change_type property of a GitDiffSummary.
    #: This constant has a value of "UNMERGED"
    CHANGE_TYPE_UNMERGED = "UNMERGED"

    #: A constant which can be used with the conflict_type property of a GitDiffSummary.
    #: This constant has a value of "BOTH_MODIFIED"
    CONFLICT_TYPE_BOTH_MODIFIED = "BOTH_MODIFIED"

    #: A constant which can be used with the conflict_type property of a GitDiffSummary.
    #: This constant has a value of "BOTH_ADDED"
    CONFLICT_TYPE_BOTH_ADDED = "BOTH_ADDED"

    #: A constant which can be used with the conflict_type property of a GitDiffSummary.
    #: This constant has a value of "BOTH_DELETED"
    CONFLICT_TYPE_BOTH_DELETED = "BOTH_DELETED"

    #: A constant which can be used with the conflict_type property of a GitDiffSummary.
    #: This constant has a value of "ADDED_BY_US"
    CONFLICT_TYPE_ADDED_BY_US = "ADDED_BY_US"

    #: A constant which can be used with the conflict_type property of a GitDiffSummary.
    #: This constant has a value of "ADDED_BY_THEM"
    CONFLICT_TYPE_ADDED_BY_THEM = "ADDED_BY_THEM"

    #: A constant which can be used with the conflict_type property of a GitDiffSummary.
    #: This constant has a value of "DELETED_BY_US"
    CONFLICT_TYPE_DELETED_BY_US = "DELETED_BY_US"

    #: A constant which can be used with the conflict_type property of a GitDiffSummary.
    #: This constant has a value of "DELETED_BY_THEM"
    CONFLICT_TYPE_DELETED_BY_THEM = "DELETED_BY_THEM"

    #: A constant which can be used with the conflict_type property of a GitDiffSummary.
    #: This constant has a value of "UNKNOWN"
    CONFLICT_TYPE_UNKNOWN = "UNKNOWN"

    def __init__(self, **kwargs):
        """
        Initializes a new GitDiffSummary object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param git_file_path:
            The value to assign to the git_file_path property of this GitDiffSummary.
        :type git_file_path: str

        :param change_type:
            The value to assign to the change_type property of this GitDiffSummary.
            Allowed values for this property are: "ADDED", "MODIFIED", "DELETED", "RENAMED", "COPIED", "TYPE_CHANGED", "UNMERGED", 'UNKNOWN_ENUM_VALUE'.
            Any unrecognized values returned by a service will be mapped to 'UNKNOWN_ENUM_VALUE'.
        :type change_type: str

        :param old_git_file_path:
            The value to assign to the old_git_file_path property of this GitDiffSummary.
        :type old_git_file_path: str

        :param is_conflict:
            The value to assign to the is_conflict property of this GitDiffSummary.
        :type is_conflict: bool

        :param conflict_type:
            The value to assign to the conflict_type property of this GitDiffSummary.
            Allowed values for this property are: "BOTH_MODIFIED", "BOTH_ADDED", "BOTH_DELETED", "ADDED_BY_US", "ADDED_BY_THEM", "DELETED_BY_US", "DELETED_BY_THEM", "UNKNOWN", 'UNKNOWN_ENUM_VALUE'.
            Any unrecognized values returned by a service will be mapped to 'UNKNOWN_ENUM_VALUE'.
        :type conflict_type: str

        """
        self.swagger_types = {
            'git_file_path': 'str',
            'change_type': 'str',
            'old_git_file_path': 'str',
            'is_conflict': 'bool',
            'conflict_type': 'str'
        }

        self.attribute_map = {
            'git_file_path': 'gitFilePath',
            'change_type': 'changeType',
            'old_git_file_path': 'oldGitFilePath',
            'is_conflict': 'isConflict',
            'conflict_type': 'conflictType'
        }

        self._git_file_path = None
        self._change_type = None
        self._old_git_file_path = None
        self._is_conflict = None
        self._conflict_type = None

    @property
    def git_file_path(self):
        """
        **[Required]** Gets the git_file_path of this GitDiffSummary.
        File path relative to repository root.


        :return: The git_file_path of this GitDiffSummary.
        :rtype: str
        """
        return self._git_file_path

    @git_file_path.setter
    def git_file_path(self, git_file_path):
        """
        Sets the git_file_path of this GitDiffSummary.
        File path relative to repository root.


        :param git_file_path: The git_file_path of this GitDiffSummary.
        :type: str
        """
        self._git_file_path = git_file_path

    @property
    def change_type(self):
        """
        **[Required]** Gets the change_type of this GitDiffSummary.
        Type of change for this file.

        Allowed values for this property are: "ADDED", "MODIFIED", "DELETED", "RENAMED", "COPIED", "TYPE_CHANGED", "UNMERGED", 'UNKNOWN_ENUM_VALUE'.
        Any unrecognized values returned by a service will be mapped to 'UNKNOWN_ENUM_VALUE'.


        :return: The change_type of this GitDiffSummary.
        :rtype: str
        """
        return self._change_type

    @change_type.setter
    def change_type(self, change_type):
        """
        Sets the change_type of this GitDiffSummary.
        Type of change for this file.


        :param change_type: The change_type of this GitDiffSummary.
        :type: str
        """
        allowed_values = ["ADDED", "MODIFIED", "DELETED", "RENAMED", "COPIED", "TYPE_CHANGED", "UNMERGED"]
        if not value_allowed_none_or_none_sentinel(change_type, allowed_values):
            change_type = 'UNKNOWN_ENUM_VALUE'
        self._change_type = change_type

    @property
    def old_git_file_path(self):
        """
        Gets the old_git_file_path of this GitDiffSummary.
        Previous path if the file was renamed.


        :return: The old_git_file_path of this GitDiffSummary.
        :rtype: str
        """
        return self._old_git_file_path

    @old_git_file_path.setter
    def old_git_file_path(self, old_git_file_path):
        """
        Sets the old_git_file_path of this GitDiffSummary.
        Previous path if the file was renamed.


        :param old_git_file_path: The old_git_file_path of this GitDiffSummary.
        :type: str
        """
        self._old_git_file_path = old_git_file_path

    @property
    def is_conflict(self):
        """
        **[Required]** Gets the is_conflict of this GitDiffSummary.
        Whether the file is currently in an unmerged/conflicted state.


        :return: The is_conflict of this GitDiffSummary.
        :rtype: bool
        """
        return self._is_conflict

    @is_conflict.setter
    def is_conflict(self, is_conflict):
        """
        Sets the is_conflict of this GitDiffSummary.
        Whether the file is currently in an unmerged/conflicted state.


        :param is_conflict: The is_conflict of this GitDiffSummary.
        :type: bool
        """
        self._is_conflict = is_conflict

    @property
    def conflict_type(self):
        """
        Gets the conflict_type of this GitDiffSummary.
        Conflict classification when isConflict is true.

        Allowed values for this property are: "BOTH_MODIFIED", "BOTH_ADDED", "BOTH_DELETED", "ADDED_BY_US", "ADDED_BY_THEM", "DELETED_BY_US", "DELETED_BY_THEM", "UNKNOWN", 'UNKNOWN_ENUM_VALUE'.
        Any unrecognized values returned by a service will be mapped to 'UNKNOWN_ENUM_VALUE'.


        :return: The conflict_type of this GitDiffSummary.
        :rtype: str
        """
        return self._conflict_type

    @conflict_type.setter
    def conflict_type(self, conflict_type):
        """
        Sets the conflict_type of this GitDiffSummary.
        Conflict classification when isConflict is true.


        :param conflict_type: The conflict_type of this GitDiffSummary.
        :type: str
        """
        allowed_values = ["BOTH_MODIFIED", "BOTH_ADDED", "BOTH_DELETED", "ADDED_BY_US", "ADDED_BY_THEM", "DELETED_BY_US", "DELETED_BY_THEM", "UNKNOWN"]
        if not value_allowed_none_or_none_sentinel(conflict_type, allowed_values):
            conflict_type = 'UNKNOWN_ENUM_VALUE'
        self._conflict_type = conflict_type

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/GitDiffSummary.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;


/**
 * Summary information about a file that differs in a Git folder context.
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=GitDiffSummary.Builder.class)

public final class GitDiffSummary  {
    @Deprecated
    @java.beans.ConstructorProperties({"gitFilePath", "changeType", "oldGitFilePath", "isConflict", "conflictType"})
    public GitDiffSummary(String gitFilePath, ChangeType changeType, String oldGitFilePath, Boolean isConflict, ConflictType conflictType) {
        super();
        this.gitFilePath = gitFilePath;
        this.changeType = changeType;
        this.oldGitFilePath = oldGitFilePath;
        this.isConflict = isConflict;
        this.conflictType = conflictType;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            /**
     * File path relative to repository root.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("gitFilePath")
        private String gitFilePath;

                /**
         * File path relative to repository root.
         * @param gitFilePath the value to set
         * @return this builder
         **/
        

        public Builder gitFilePath(String gitFilePath) {
        this.gitFilePath = gitFilePath;
        return this;
        }
            /**
     * Type of change for this file.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("changeType")
        private ChangeType changeType;

                /**
         * Type of change for this file.
         * @param changeType the value to set
         * @return this builder
         **/
        

        public Builder changeType(ChangeType changeType) {
        this.changeType = changeType;
        return this;
        }
            /**
     * Previous path if the file was renamed.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("oldGitFilePath")
        private String oldGitFilePath;

                /**
         * Previous path if the file was renamed.
         * @param oldGitFilePath the value to set
         * @return this builder
         **/
        

        public Builder oldGitFilePath(String oldGitFilePath) {
        this.oldGitFilePath = oldGitFilePath;
        return this;
        }
            /**
     * Whether the file is currently in an unmerged/conflicted state.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("isConflict")
        private Boolean isConflict;

                /**
         * Whether the file is currently in an unmerged/conflicted state.
         * @param isConflict the value to set
         * @return this builder
         **/
        

        public Builder isConflict(Boolean isConflict) {
        this.isConflict = isConflict;
        return this;
        }
            /**
     * Conflict classification when isConflict is true.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("conflictType")
        private ConflictType conflictType;

                /**
         * Conflict classification when isConflict is true.
         * @param conflictType the value to set
         * @return this builder
         **/
        

        public Builder conflictType(ConflictType conflictType) {
        this.conflictType = conflictType;
        return this;
        }


        public GitDiffSummary build() {
            GitDiffSummary model = new GitDiffSummary(this.gitFilePath
                    , this.changeType
                    , this.oldGitFilePath
                    , this.isConflict
                    , this.conflictType);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(GitDiffSummary model) {
                this.gitFilePath(model.getGitFilePath());
    this.changeType(model.getChangeType());
    this.oldGitFilePath(model.getOldGitFilePath());
    this.isConflict(model.getIsConflict());
    this.conflictType(model.getConflictType());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * File path relative to repository root.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("gitFilePath")
    private final String gitFilePath;

        /**
     * File path relative to repository root.
     * @return the value
     **/
    
    public String getGitFilePath() {
        return gitFilePath;
    }

    /**
     * Type of change for this file.
     **/
    public enum ChangeType {
        Added("ADDED"),
        Modified("MODIFIED"),
        Deleted("DELETED"),
        Renamed("RENAMED"),
        Copied("COPIED"),
        TypeChanged("TYPE_CHANGED"),
        Unmerged("UNMERGED"),
        

        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        UnknownEnumValue(null);

        private static final org.slf4j.Logger LOG = org.slf4j.LoggerFactory.getLogger(ChangeType.class);

        private final String value;
        private static java.util.Map<String, ChangeType> map;

        static {
            map = new java.util.HashMap<>();
            for (ChangeType v : ChangeType.values()) {
                if (v != UnknownEnumValue) {
                    map.put(v.getValue(), v);
                }
            }
        }

        ChangeType(String value) {
            this.value = value;
        }

        @com.fasterxml.jackson.annotation.JsonValue
        public String getValue() {
            return value;
        }

        @com.fasterxml.jackson.annotation.JsonCreator
        public static ChangeType create(String key) {
            if (map.containsKey(key)) {
                return map.get(key);
            }
            LOG.warn("Received unknown value '{}' for enum 'ChangeType', returning UnknownEnumValue", key);
            return UnknownEnumValue;
        }
    };
        /**
     * Type of change for this file.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("changeType")
    private final ChangeType changeType;

        /**
     * Type of change for this file.
     * @return the value
     **/
    
    public ChangeType getChangeType() {
        return changeType;
    }


        /**
     * Previous path if the file was renamed.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("oldGitFilePath")
    private final String oldGitFilePath;

        /**
     * Previous path if the file was renamed.
     * @return the value
     **/
    
    public String getOldGitFilePath() {
        return oldGitFilePath;
    }


        /**
     * Whether the file is currently in an unmerged/conflicted state.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("isConflict")
    private final Boolean isConflict;

        /**
     * Whether the file is currently in an unmerged/conflicted state.
     * @return the value
     **/
    
    public Boolean getIsConflict() {
        return isConflict;
    }

    /**
     * Conflict classification when isConflict is true.
     **/
    public enum ConflictType {
        BothModified("BOTH_MODIFIED"),
        BothAdded("BOTH_ADDED"),
        BothDeleted("BOTH_DELETED"),
        AddedByUs("ADDED_BY_US"),
        AddedByThem("ADDED_BY_THEM"),
        DeletedByUs("DELETED_BY_US"),
        DeletedByThem("DELETED_BY_THEM"),
        Unknown("UNKNOWN"),
        

        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        UnknownEnumValue(null);

        private static final org.slf4j.Logger LOG = org.slf4j.LoggerFactory.getLogger(ConflictType.class);

        private final String value;
        private static java.util.Map<String, ConflictType> map;

        static {
            map = new java.util.HashMap<>();
            for (ConflictType v : ConflictType.values()) {
                if (v != UnknownEnumValue) {
                    map.put(v.getValue(), v);
                }
            }
        }

        ConflictType(String value) {
            this.value = value;
        }

        @com.fasterxml.jackson.annotation.JsonValue
        public String getValue() {
            return value;
        }

        @com.fasterxml.jackson.annotation.JsonCreator
        public static ConflictType create(String key) {
            if (map.containsKey(key)) {
                return map.get(key);
            }
            LOG.warn("Received unknown value '{}' for enum 'ConflictType', returning UnknownEnumValue", key);
            return UnknownEnumValue;
        }
    };
        /**
     * Conflict classification when isConflict is true.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("conflictType")
    private final ConflictType conflictType;

        /**
     * Conflict classification when isConflict is true.
     * @return the value
     **/
    
    public ConflictType getConflictType() {
        return conflictType;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("GitDiffSummary(");
        sb.append("gitFilePath=").append(String.valueOf(this.gitFilePath));
        sb.append(", changeType=").append(String.valueOf(this.changeType));
        sb.append(", oldGitFilePath=").append(String.valueOf(this.oldGitFilePath));
        sb.append(", isConflict=").append(String.valueOf(this.isConflict));
        sb.append(", conflictType=").append(String.valueOf(this.conflictType));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof GitDiffSummary)) {
            return false;
        }

        GitDiffSummary other = (GitDiffSummary) o;
        return java.util.Objects.equals(this.gitFilePath, other.gitFilePath) &&
            java.util.Objects.equals(this.changeType, other.changeType) &&
            java.util.Objects.equals(this.oldGitFilePath, other.oldGitFilePath) &&
            java.util.Objects.equals(this.isConflict, other.isConflict) &&
            java.util.Objects.equals(this.conflictType, other.conflictType);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.gitFilePath == null ? 43 : this.gitFilePath.hashCode());
        result = (result * PRIME) + (this.changeType == null ? 43 : this.changeType.hashCode());
        result = (result * PRIME) + (this.oldGitFilePath == null ? 43 : this.oldGitFilePath.hashCode());
        result = (result * PRIME) + (this.isConflict == null ? 43 : this.isConflict.hashCode());
        result = (result * PRIME) + (this.conflictType == null ? 43 : this.conflictType.hashCode());
        return result;
    }


}
